### PR TITLE
remove ConcurrentMapChunkCache

### PR DESCRIPTION
**Pull request by pollend**
I removed the ConcurrentMapChunkCache since it just seems to just wrap a map unless we have some plans to support different types of cache. the other thing we can do is Also use the same implementation for the RemoteChunkProvider since that implementation also uses a chunkCache and that just uses a Maps.newHashMap(). At the moment there doesn't really seem to be a need to implement more then one type of cache. 

For chunkFinalizer I removed the instance that gets passed in at startup and just grab an instance out of the Supplier. when purgeWorld is called; this will trigger another instance from the Supplier. this would make sense since the only other instance just declares new LightMergingChunkFinalizer and LightMergingChunkFinalizer::new.

### CODE REVIEW

File: engine/src/main/java/org/terasology/world/chunks/localChunkProvider/LocalChunkProvider.java [hunk 1]
```diff
@@ -112,38 +111,23 @@
 
     private BlockManager blockManager;
     private ExtraBlockDataManager extraDataManager;
-    private final ChunkCache chunkCache;
-    private final Supplier<ChunkFinalizer> chunkFinalizerSupplier;
+    private final Map<Vector3i, Chunk> chunkCache = Maps.newConcurrentMap();
```
Comment (skaldarnar) [inline engine/src/main/java/org/terasology/world/chunks/localChunkProvider/LocalChunkProvider.java:114]: I guess this needs to be JOML-ified now?

File: engine/src/main/java/org/terasology/world/chunks/localChunkProvider/LocalChunkProvider.java [hunk 2]
```diff
@@ -112,38 +111,23 @@
...
     private BlockEntityRegistry registry;
 
+    //TODO: Michael Pollind -- if we have more then one finalizer use  Supplier<ChunkFinalizer[]> and ChunkFinalizer[]
+    private final Supplier<ChunkFinalizer> chunkFinalizerSupplier;
     private ChunkFinalizer chunkFinalizer;
 
-    //TODO Remove this old constructor at the end of the chunk overhaul
```
Comment (skaldarnar) [inline engine/src/main/java/org/terasology/world/chunks/localChunkProvider/LocalChunkProvider.java:121]: when is the "end of the chunk overhaul"? 🤔

File: engine/src/main/java/org/terasology/world/chunks/localChunkProvider/LocalChunkProvider.java [hunk 3]
```diff
@@ -112,38 +111,23 @@
 
     private BlockManager blockManager;
     private ExtraBlockDataManager extraDataManager;
-    private final ChunkCache chunkCache;
-    private final Supplier<ChunkFinalizer> chunkFinalizerSupplier;
+    private final Map<Vector3i, Chunk> chunkCache = Maps.newConcurrentMap();
     private BlockEntityRegistry registry;
 
+    //TODO: Michael Pollind -- if we have more then one finalizer use  Supplier<ChunkFinalizer[]> and ChunkFinalizer[]
+    private final Supplier<ChunkFinalizer> chunkFinalizerSupplier;
```
Comment (skaldarnar) [inline engine/src/main/java/org/terasology/world/chunks/localChunkProvider/LocalChunkProvider.java:118]: What do we need to store the supplier as member here? When is it used?

**Review by skaldarnar — COMMENTED** [review]
Hey @pollend, is this still relevant, or rather, should we try to get this merged? I'm always a fan of removing code, so belated 👍 from here 😉

**Comment by DarkWeird** [comment]
This PR not actual after ChunkRefactoring part 2 :p

**Comment by pollend** [comment]
I can re-base these changes on to the latest develop.

**Comment by pollend** [comment]
@DarkWeird there you go. looks like we can just remove ConcurrentMapChunkCache

**Review by DarkWeird — APPROVED** [review]

**Pull request merged**